src/modules/one-signal-client/one-signal-client.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, Logger, BadGatewayException, Head } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { AxiosError } from 'axios';
+import { catchError, retry } from 'rxjs';
+import { UserDevice } from '../price-notifications/entity/user-device.entity';
+import { TokenPrice } from '../token-price/entity/token-price.entity';
+
+const ONE_SIGNAL_API = 'ONE_SIGNAL_API';
+const ONE_SIGNAL_APP_ID = 'ONE_SIGNAL_APP_ID';
+const ONE_SIGNAL_API_KEY = 'ONE_SIGNAL_API_KEY';
+
+@Injectable()
+export class OneSignalClient {
+  private readonly logger = new Logger(OneSignalClient.name);
+
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly configs: ConfigService,
+  ) {}
+
+  async sendPriceChangeNotification(users: UserDevice[], token: TokenPrice) {
+    const userIds = users.map((user) => user.deviceId);
+    if (userIds.length === 0) return;
```

Comment:
This approach is more readable and less error prone, each if block have it's parentheses. 
```suggestion
    if (userIds.length === 0) {
      return;
    }
```